File: string_analyzer.py
def analyze_string(input_string):
    num_characters = len(input_string)
    word_list = input_string.split()
    num_words = len(word_list)
    
    # Cleaning the input: remove non-alphabetic characters and convert to lowercase
    cleaned_input = ''.join(char.lower() for char in input_string if char.isalpha())
    
    clean_length = len(cleaned_input)
    
    # Counting vowels and consonants
    vowels = "aeiou"
    num_vowels = sum(1 for char in cleaned_input if char in vowels)
    num_consonants = clean_length - num_vowels

    result_dict = {
        "num_characters": num_characters,
        "num_words": num_words,
        "num_vowels": num_vowels,
        "num_consonants": num_consonants
    }
    
    return result_dict

if __name__ == "__main__":
    input_string = "Your sample input string goes here."
    print(analyze_string(input_string))

File: stringanalyzer.js
function analyzeString(inputString) {
    const numCharacters = inputString.length;
    const wordList = inputString.split(/\s+/); // Split by whitespace
    const numWords = wordList.length;

    // Cleaning the input: remove non-alphabetic characters and convert to lowercase
    const cleanedInput = inputString.replace(/[^a-zA-Z]/g, '').toLowerCase();

    const cleanLength = cleanedInput.length;

    // Counting vowels and consonants
    const vowels = 'aeiou';
    let numVowels = 0;
    for (let char of cleanedInput) {
        if (vowels.includes(char)) {
            numVowels++;
        }
    }
    const numConsonants = cleanLength - numVowels;

    const resultDict = {
        numCharacters: numCharacters,
        numWords: numWords,
        numVowels: numVowels,
        numConsonants: numConsonants
    };

    return resultDict;
}

const inputString = "Your sample input string goes here.";
console.log(analyzeString(inputString));
